core/src/main/scala/io/finch/syntax/EndpointMapper.scala

```diff
@@ -8,7 +8,7 @@ class EndpointMapper[A](m: Method, e: Endpoint[A]) extends Endpoint[A] { self =>
   /**
    * Maps this endpoint to either `A => Output[B]` or `A => Future[Output[B]]`.
    */
-  final def apply(mapper: Mapper[A]): Endpoint[mapper.Out] = mapper(self)
+  final def apply[F](f: => F)(implicit mapper: Mapper[F, A]): Endpoint[mapper.Out] = mapper(f, self)
```

Comment:
Spent a day, trying to find possible workaround. I wish we would have Dotty now -__-
Seems like magnet is the only possible choice here if there is a need of that kind of optimization. I could move `ToTwitterFuture` stuff to different PR